### PR TITLE
Feature/basic UI

### PR DESCRIPTION
**Pull request by ClaudiaRojasSoto**
# Description:
This pull request introduces a new feature, Basic-UI, to enhance the school library app by providing a user-friendly form of interaction. The primary goal of this feature is to allow the app to be invoked as an executable, enhancing its usability beyond IRB.

## Changes Made:
- [x] Create a user interface for the school library app, enabling users to interact with the app in a more intuitive manner.
- [x] Implement the UI based on the example provided in the video, ensuring consistency and familiarity for users.

### General Requirements:
- [x] Ensure the code adheres to linting standards, with no linter errors or warnings present.
- [x] Follow Gitflow methodology, maintaining separate branches for feature development, and using proper branching conventions.
- [x] Document the implementation process professionally, providing clear and concise comments within the codebase.
- [x] Adhere to the list of best practices for Ruby programming, maintaining clean and readable code.

### Project Requirements:
- [x] Watch the UI example video to understand the desired behavior and user experience.
- [x] Create an `app.rb` file to serve as the entry point for the console app, containing methods as per the specified requirements.
- [x] Implement methods to list books, list people, create a person (teacher or student), create a book, create a rental, and list all rentals for a given person id.
- [x] Define the main method in `main.rb` as the entry point for user interaction.
- [x] Present users with a menu of options, allowing them to choose actions to perform.
- [x] Prompt users for any necessary parameters required for each action.
- [x] Provide a way for users to quit the app gracefully.

This pull request represents a significant enhancement to the school library app, bringing a user-friendly interface to its functionality. The code has been reviewed and tested to meet the stipulated requirements, ensuring a seamless integration of the Basic-UI feature into the develop branch. Please review this feature and give me feedback in order to improve it.
Greetings,
@ClaudiaRojasSoto

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      puts 'Student created successfully'
+    when '2'
+      teacher = create_teacher
+      @people << teacher
+      puts 'Teacher created successfully'
+    else
+      puts 'Invalid option'
+    end
+  end
+
```
Comment (Reem-lab) [inline app.rb:42]: - [x] Great job working on the functionality so far but when I try to create a new person I face this error wrong number of arguments (given 4, expected 3) so kindly check on this error and try to solve the problem to let the user be able to create a person whether a student or a teacher ⭐🔥💯🙏🏻

- When I try to create a student  ⬇
image

- When I try to create a teacher⬇
image

**Review by Reem-lab — CHANGES_REQUESTED** [review]
# Hi @ClaudiaRojasSoto, 👋🏻👋🏻

Great job so far! ✨⭐🤩✅
There are some issues that you still need to work on to go to the next project but you are almost there!
### Note to the next reviewer: kindly check on the rental if it is working fine ✅ The book is working well 🎉



## Highlights
- [ ] No linters error ✅
- [ ] Followed Gitflow ⭐
- [ ] Great job working on the functionality 💯✨
- [ ] Descriptive PR 🚀

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (ClaudiaRojasSoto) [reply app.rb:42]: Thank you @Reem-lab. I solved the problems and now the app is 100% functional.

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🙌

Hi @ClaudiaRojasSoto 

Good job implementing the suggested changes💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Great job with the UI
-  ✅ Great work using the right GitFlow 
-  ✅ Linters are working properly

## Optional suggestions ❕
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ClaudiaRojasSoto** [comment]
Thank you @KanzaTahreem.

**Pull request merged**